.github/workflows/linux.yml

```diff
@@ -27,7 +27,8 @@ jobs:
         restore-keys: |
           ${{ runner.os }}-maven-
     - name: Build with Maven
-      run: mvn -B clean install -Dspotless.apply.skip=true -Dall -T2 --file geowebcache/pom.xml
+      # -DskipITs skips failsafe integration tests but runs unit tests with surefire. Integration tests have their own jobs
+      run: mvn -B clean install -Dspotless.apply.skip=true -Dall -T2 --file geowebcache/pom.xml -DskipITs
```

Comment:
The practice in GT/GS/GWC is that online tests are run only when explicitly required. Please don't introduce an opposite practice with a different flag name to enable it (option B, discuss this new approach on list and then apply uniformly in all 3 projects).

Reply:
but for the traditional `*OnlineTest.java` tests, "explicitly required" means the required `.properties` file is in place.
In gt/gs/gwc, so far nobody uses the failsafe plugin to run integration tests (except gs' `community/rest-openapi/pom.xml` and gt's `modules/unsupported/elasticsearch`.

My point of view is if you wrote `*IT.java` integration tests to be run by failsafe, it is because you expect them to run after the unit tests succeeded and after the maven package phase. And hopefully as often as the unit tests (e.g. during development).

So introducing `-DskipITs` here makes sense cause we want to preserve the traditional behavior [*], and setup specific github actions to run ITs for specific modules (e.g. I'd like to do the same for the S3 blobstore), but still the default action would be to run unit then integration tests if you do `mvn clean install` or `mvn verify` during development.

> \* i.e., run both tests and online tests with surefire

Reply:
I see that you're going towards the maven default behavior, which is not bad per-se.
But it's also something that should be discussed on the list (geotools-devel or geoserver-devel). Having slow-to-execute tests run by default is not our default, the community should decide if it's desirable to start with, and how to deal with the (temporary?) inconsistency between existing online tests and the newly introduced integration tests.

Reply:
I've no trouble disabling failsafe by default. What you propose would be a separate discussion than the one for this pull request, and I'm afraid I don't have the bandwidth to engage right now. I'd rather spend my time on the actual migration to the new libraries, since it'll also imply doing to at the imageio-ext side of the fence.

But not as it stands, in order to run the traditional "online" tests for Azure, it won't be enough to have the .properties file in place, you'll also need to specify `-DskipITs=false`, since I've refactored them to run during the integration-tests phase. Otherwise you'd be running the online tests at the same time than the unit tests, and hopefully one should avoid that.

Is that a good compromise?

Reply:
@groldan in our geo-trio projects we always highlighted the importance of communicating before coding, I see you assuming that what you're doing is "obviously better and thus requires no discussion" too often. 
Since you don't want to have preparatory discussions on the list, we're going to have at least one here, and it can't be just me and you, it needs a few more interested parties.

Since it's PSC meeting day, I'll add the usage of the failsafe plugin as a topic, again for the sake of having a larger audience.

@jodygarnett @smithkm can you please share your opinion here? Personally I'm happy with whatever consistent direction, as long as there is broad agreement on the way to go.

Reply:
I've run the build, the ITs add a couple of minutes to the Azure module build.

In terms of controlling the test execution, an existing alternative example s the ElasticSearch tests in GeoTools, which uses failsafe and the IT naming convention, does not need a property file, but has the failsafe execution enabled [only when the online profile is enabled](https://github.com/geotools/geotools/blob/main/modules/unsupported/elasticsearch/pom.xml#L197).

Reply:
>@groldan in our geo-trio projects we always highlighted the importance of communicating before coding, I see you assuming that what you're doing is "obviously better and thus requires no discussion" too often.

In this particular case, running the integration tests after the unit tests is not only "obviously better" than not, but also standard practice in the industry.

As for your general statement about my behavior, I strongly disagree and I'm sorry you feel that way.

> Since you don't want to have preparatory discussions on the list, we're going to have at least one here, and it can't be just me and you, it needs a few more interested parties.

A pull request by definition is a proposal for a change, and an opportunity to discuss with the bulk of the work done in advance, and hence the confidence that the proposed approach works, as opposed to discussing in the air.
The pull request contents are way more important than the discussion of whether we run the integration tests by default or not (given I've solved a problem that lasted for years) , for which I agreed to revert to usual behavior, but asked if you're ok with running the online tests when explicitly asked for. The mechanics of it could be through `-DskipITs` or `-Ponline`, but I didn't think the practice on a community module would be any indication of a standard practice.

> I've run the build, the ITs add a couple of minutes to the Azure module build.
I don't get if that's a complain or just a statement

Reply:
`-Ponline` it is, as discussed in the PSC meeting. I'll do that.